include/openPMD/IO/ADIOS/ADIOS1Auxiliary.hpp

```diff
@@ -87,23 +87,23 @@ getBP1DataType(Datatype dtype)
         case DT::VEC_UCHAR:
         case DT::BOOL:
             return adios_unsigned_byte;
-        case DT::INT16:
-        case DT::VEC_INT16:
+        case DT::SHORT:
+        case DT::VEC_SHORT:
             return adios_short;
-        case DT::INT32:
-        case DT::VEC_INT32:
+        case DT::INT:
+        case DT::VEC_INT:
             return adios_integer;
-        case DT::INT64:
-        case DT::VEC_INT64:
+        case DT::LONG:
+        case DT::VEC_LONG:
             return adios_long;
-        case DT::UINT16:
-        case DT::VEC_UINT16:
+        case DT::USHORT:
+        case DT::VEC_USHORT:
             return adios_unsigned_short;
-        case DT::UINT32:
-        case DT::VEC_UINT32:
+        case DT::UINT:
+        case DT::VEC_UINT:
             return adios_unsigned_integer;
-        case DT::UINT64:
-        case DT::VEC_UINT64:
+        case DT::ULONG:
+        case DT::VEC_ULONG:
```

Comment:
We're very likely going to have problems with ADIOS in this regard.
From what I can see, they do **NOT** make the destinction that causes problems with OSX and MSVC. Rather, they treat the bit-widths as I did initally (see [```adios_types.h```](https://github.com/ornladios/ADIOS/blob/v1.13.1/src/public/adios_types.h#L15-L39), which explicitly lists sizes of the datatypes).
As e.g. `int` may now be either 16 or 32 bit, and ADIOS assumes a size of 32, this might explode in our face.

Reply:
Yep, I know but that's ADIOS portability problem not ours: https://github.com/ornladios/ADIOS/issues/187

Reply:
Uh wait, you mean we will not be able to store a `(u)int64_t` since ADIOS has no `long long`... Yeah... Well.

Reply:
> that's ADIOS' portability problem not ours

So the baseline here is to rely on the C standard and ignore potentially faulty behaviour in one of our backends. Not ideal, but probably the only thing we can do.

> you mean we will not be able to store a (u)int64_t

We most likely can, unless we're on exotic platforms (*MSVC*, anyone?) where `LONG` and `ULONG` are 32 Bits wide... Which again comes back to the standard compliance stated above.

Reply:
What I try to say is: if `adios_long` is defined as 8 byte on *all* platforms, we will implement it as such and convert properly in the backend from whatever matches (`int` or `long` or `long long`).

Reply:
>The fileformat is responsible for portability, e.g. HDF5 stores automatically platform information with the stored types

Hi, just so I get this right: When reading e.g. a `long` from some stored file, I will also have to take into account the platform that wrote the value and use the reading platform's datatype that corresponds with the actual length of the value stored, so possibly not a `long` but an `int` for example?
If so, it's good you mention this, because floating point datatypes already have the same issue and I ignored that so far.

Reply:
> I will also have to take into account the platform

My idea is to handle it exactly the other way around. Our backends, such as HDF5, are portable across platforms. If HDF5 stores for example a `long` (aka `int32_t`) on OSX it does store in the back automatically the type of the platform. If you read this file back on x86-64 Linux, the HDF type presented to you will be `int` (still aka `int32_t`) since `long` (and `long long`) are `int64_t` there. The precision of the data is preserved.

> JSON

Let's migrate that discussion here: https://github.com/openPMD/openPMD-api/issues/65